### PR TITLE
kvutils: Remove long unapplies for transaction `Node`s

### PR DESCRIPTION
**Pull request by miklos-da**
Motivation: reduce code churn because of changing constructors for these types.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala [hunk 1]
```diff
@@ -627,6 +624,7 @@ private[kvutils] class TransactionCommitter(
         .map(v => v.getNumber -> metrics.daml.kvutils.committer.transaction.rejection(v.name()))
         .toMap
   }
+
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala:627]: I don't see this NL as necessary

**Review by fabiotudone-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 2]
```diff
@@ -79,23 +82,28 @@ private[kvutils] object InputsAndEffects {
     tx.foreach {
       case (_, node) =>
         node match {
-          case fetch @ Node.NodeFetch(_, _, _, _, _, _, _, _) =>
+          case fetch: Node.NodeFetch[Value.ContractId, Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:85]:
```suggestion
          case fetch: Node.NodeFetch.WithTxValue[Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 3]
```diff
@@ -79,23 +82,28 @@ private[kvutils] object InputsAndEffects {
...
             addContractInput(fetch.coid)
             fetch.key.foreach { keyWithMaintainers =>
               inputs += globalKeyToStateKey(
                 GlobalKey(fetch.templateId, forceNoContractIds(keyWithMaintainers.key.value)))
             }
 
-          case create @ Node.NodeCreate(_, _, _, _, _, _) =>
+          case create: Node.NodeCreate[Value.ContractId, Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:92]:
```suggestion
          case create: Node.NodeCreate.Node.NodeFetch.WithTxValue[Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 4]
```diff
@@ -79,23 +82,28 @@ private[kvutils] object InputsAndEffects {
...
             create.key.foreach { keyWithMaintainers =>
               inputs += globalKeyToStateKey(
                 GlobalKey(create.coinst.template, forceNoContractIds(keyWithMaintainers.key.value)))
             }
 
-          case exe @ Node.NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _) =>
+          case exe: Node.NodeExercises[
+                NodeId,
+                Value.ContractId,
+                Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:101]:
```suggestion
          case exe: Node.NodeExercises.WithTxValue[NodeId,Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 5]
```diff
@@ -79,23 +82,28 @@ private[kvutils] object InputsAndEffects {
...
             addContractInput(exe.targetCoid)
 
-          case lookup @ Node.NodeLookupByKey(_, _, _, _) =>
+          case lookup: Node.NodeLookupByKey[
+                Value.ContractId,
+                Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:106]:
```suggestion
          case lookup: Node.NodeLookupByKey.WithTxValue[Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 6]
```diff
@@ -114,18 +122,18 @@ private[kvutils] object InputsAndEffects {
     // TODO(JM): Skip transient contracts in createdContracts/updateContractKeys. E.g. rewrite this to
     // fold bottom up (with reversed roots!) and skip creates of archived contracts.
     tx.fold(Effects.empty) {
-      case (effects, (nodeId @ _, node)) =>
+      case (effects, (_, node)) =>
         node match {
-          case Node.NodeFetch(_, _, _, _, _, _, _, _) =>
+          case _: Node.NodeFetch[Value.ContractId, Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:127]:
```suggestion
          case _: Node.NodeFetch.WithTxValue[Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 7]
```diff
@@ -114,18 +122,18 @@ private[kvutils] object InputsAndEffects {
...
             effects
-          case create @ Node.NodeCreate(_, _, _, _, _, _) =>
+          case create: Node.NodeCreate[Value.ContractId, Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:129]:
```suggestion
          case create: Node.NodeCreate.WithTxValue[Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 8]
```diff
@@ -135,29 +143,32 @@ private[kvutils] object InputsAndEffects {
...
                       Some(create.coid)
                   )
                 )
             )
 
-          case exe @ Node.NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _) =>
+          case exe: Node.NodeExercises[
+                NodeId,
+                Value.ContractId,
+                Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:155]:
```suggestion
          case exe: Node.NodeExercises.WithTxValue[NodeId, Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 9]
```diff
@@ -135,29 +143,32 @@ private[kvutils] object InputsAndEffects {
...
+                        globalKeyToStateKey(
+                          GlobalKey(exe.templateId, forceNoContractIds(key.key.value))),
                         None)
                   )
               )
             } else {
               effects
             }
-          case Node.NodeLookupByKey(_, _, _, _) =>
+          case _: Node.NodeLookupByKey[Value.ContractId, Transaction.Value[Value.ContractId]] =>
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:171]:
```suggestion
          case _: Node.NodeLookupByKey.WithTxValue[Value.ContractId] =>
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala [hunk 10]
```diff
@@ -281,13 +276,15 @@ private[kvutils] class TransactionCommitter(
       .fold((true, keys)) {
         case (
             (allUnique, existingKeys),
-            (_, exe @ Node.NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _)))
+            (_, exe: Node.NodeExercises[NodeId, Value.ContractId, Tx.Value[Value.ContractId]]))
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala:279]:
```suggestion
            (_, exe: Node.NodeExercises.WithTxValue[NodeId, Value.ContractId]))
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala [hunk 11]
```diff
@@ -281,13 +276,15 @@ private[kvutils] class TransactionCommitter(
...
             if exe.key.isDefined && exe.consuming =>
           val stateKey = Conversions.globalKeyToStateKey(
             GlobalKey(exe.templateId, Conversions.forceNoContractIds(exe.key.get.key.value)))
           (allUnique, existingKeys - stateKey)
 
-        case ((allUnique, existingKeys), (_, create @ Node.NodeCreate(_, _, _, _, _, _)))
+        case (
+            (allUnique, existingKeys),
+            (_, create: Node.NodeCreate[Value.ContractId, Tx.Value[Value.ContractId]]))
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala:287]:
```suggestion
            (_, create: Node.NodeCreate.WithTxValue[Value.ContractId]))
```

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM. I just add minor suggestions.

**Pull request merged**